src/cmd.rs

```diff
@@ -214,7 +263,8 @@ impl Cmd {
     /// # let client = redis::Client::open("redis://127.0.0.1/").unwrap();
     /// # let con = client.get_connection().unwrap();
     /// let mut cmd = redis::cmd("SSCAN");
-    /// let mut iter : redis::Iter<isize> = cmd.arg("my_set").cursor_arg(0).iter(&con).unwrap();
+    /// let mut iter : redis::Iter<isize> =
+    ///     cmd.arg("my_set").cursor_arg(0).clone().iter(&con).unwrap();
```

Comment:
Hm, is this clone here now required?
That _could_ break current users.

Reply:
`iter` used to `clone` the command inside the method call before so I tweaked it to be more honest by making it take `self` by value instead. It is a bit more verbose but it is the only real way to remove that unnecessary overhead.

Reply:
(That is, users could write the command construction to avoid the clone by moving the command instead of cloning, but that is a bit more verbose)

Reply:
Hm, this only breaks when using `.iter()` then? Small break, but nothing too serious.

I'd still cut a release now and move that into the release afterwards and include other breaking changes there.

Reply:
There are other breaking changes in this PR as well though, so even if I revert the need for `clone` it would still need a breaking semver bump.

Reply:
And I think I'm fine with that for the next release. We can accumulate a few of the breaking changes and provide proper docs on what changed and how to use the new way.